### DISCUSSION TITLE
ESP32-DevKit","Version":"12.4.0.1 - Trouble to restart with failure

### DISCUSSION THREAD

**Discussion by dl7ata** [arendst/Tasmota#19101] [Q&A]
Hi folks,

maybe someone can help me interpreting the exception message after restarting the system via browser:

> ata/eHZM2/LWT Online
ata/eHZM2/cmnd/POWER (null)
ata/eHZM2/INFO1 {"Info1":{"Module":"ESP32-DevKit","Version":"12.4.0.1(tasmota)","FallbackTopic":"cmnd/DVES_AC2FE0_fb/","GroupTopic":"ata/tasmotas/cmnd/"}}
ata/eHZM2/INFO2 {"Info2":{"WebServerMode":"Admin","Hostname":"eHZM-B23cT","IPAddress":"192.168.1.12","IP6Global":"","IP6Local":"fe80::2291:51ff:feac:2fe0"}}
ata/eHZM2/INFO3 {"Info3":{"RestartReason":{"Exception":29,"Reason":"StoreProhibited","EPC":"40084e94","EXCVADDR":"00000000","CallChain":["40084e91","400913f5","40096f4d","40170a53","401098d2","40107106"]},"BootCount":42}}
ata/eHZM2/LWT Offline

Tasmota didn't start and is looping like above.

Thanks
Frank

**Comment by barbudor** [comment]
I would bet a flash problem
I suggest erase, reflash, reconfigure
And if happens again, trash

**Comment by HofeBY** [comment]
Try to use another Powersupply and Cable & reflash DevKit-C Device again.

**Reply by HofeBY** [reply]
which GIOP Ports are used ?

**Reply by dl7ata** [reply]
None - only RX with IR (Tasmota sml version)

**Reply by HofeBY** [reply]
None GPIO Port is not possible.

The reasen is, that some GPIO are not usable.
Pls show us your Script under Konsolen->Script konfigurieren
And a Hardcopy of Einstellungen->Vorlage konfigurieren & Gerät konfigurieren

from where do you get the SML-Version ?
If own compiled show us your changes in user_config_override.h and other changed files & which compiler you are using.

**Reply by dl7ata** [reply]
Script: 

> >D
>B
->sensor53 r
>M 1
+1,3,s,0,9600,Haus
1,77070100010800FF@1000,Zaehler in,kWh,total_in,2
1,77070100020800FF@1000,Zaehler out,kWh,total_out,2
1,77070100100700FF@1,Aktueller Verbrauch,W,Power_curr,0
#

Hardcopy:
![Tasmota_ 2023-07-13 21-13-21](https://github.com/arendst/Tasmota/assets/12610562/b01e60c2-1ae4-4a88-aa93-be22aa9352e8)

SML-Version: Tasmota 9.5.0 by Theo Arends with instructions from [here](https://tasmota.github.io/docs/Smart-Meter-Interface/)

user_config_override.h : 

> /*
  user_config_override.h - user configuration overrides my_user_config.h for Tasmota

  Copyright (C) 2021  Theo Arends

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

#ifndef _USER_CONFIG_OVERRIDE_H_
#define _USER_CONFIG_OVERRIDE_H_

/*****************************************************************************************************\
 * USAGE:
 *   To modify the stock configuration without changing the my_user_config.h file:
 *   (1) copy this file to "user_config_override.h" (It will be ignored by Git)
 *   (2) define your own settings below
 *
 ******************************************************************************************************
 * ATTENTION:
 *   - Changes to SECTION1 PARAMETER defines will only override flash settings if you change define CFG_HOLDER.
 *   - Expect compiler warnings when no ifdef/undef/endif sequence is used.
 *   - You still need to update my_user_config.h for major define USE_MQTT_TLS.
 *   - All parameters can be persistent changed online using commands via MQTT, WebConsole or Serial.
\*****************************************************************************************************/

/*
Examples :

// -- Master parameter control --------------------
#undef  CFG_HOLDER
#define CFG_HOLDER        4617                   // [Reset 1] Change this value to load SECTION1 configuration parameters to flash

// -- Setup your own Wifi settings  ---------------
#undef  STA_SSID1
#define STA_SSID1         "YourSSID"             // [Ssid1] Wifi SSID

#undef  STA_PASS1
#define STA_PASS1         "YourWifiPassword"     // [Password1] Wifi password

// -- Setup your own MQTT settings  ---------------
#undef  MQTT_HOST
#define MQTT_HOST         "your-mqtt-server.com" // [MqttHost]

#undef  MQTT_PORT
#define MQTT_PORT         1883                   // [MqttPort] MQTT port (10123 on CloudMQTT)

#undef  MQTT_USER
#define MQTT_USER         "YourMqttUser"         // [MqttUser] Optional user

#undef  MQTT_PASS
#define MQTT_PASS         "YourMqttPass"         // [MqttPassword] Optional password

// You might even pass some parameters from the command line ----------------------------
// Ie:  export PLATFORMIO_BUILD_FLAGS='-DUSE_CONFIG_OVERRIDE -DMY_IP="192.168.1.99" -DMY_GW="192.168.1.1" -DMY_DNS="192.168.1.1"'

#ifdef MY_IP
#undef  WIFI_IP_ADDRESS
#define WIFI_IP_ADDRESS     MY_IP                // Set to 0.0.0.0 for using DHCP or enter a static IP address
#endif

#ifdef MY_GW
#undef  WIFI_GATEWAY
#define WIFI_GATEWAY        MY_GW                // if not using DHCP set Gateway IP address
#endif

#ifdef MY_DNS
#undef  WIFI_DNS
#define WIFI_DNS            MY_DNS               // If not using DHCP set DNS IP address (might be equal to WIFI_GATEWAY)
#endif

#ifdef MY_DNS2
#undef  WIFI_DNS2
#define WIFI_DNS2           MY_DNS2              // If not using DHCP set DNS IP address (might be equal to WIFI_GATEWAY)
#endif

// !!! Remember that your changes GOES AT THE BOTTOM OF THIS FILE right before the last #endif !!!
*/

#ifndef USE_SCRIPT
#define USE_SCRIPT
#endif
#ifndef USE_SML_M
#define USE_SML_M
#endif
#ifdef USE_RULES
#undef USE_RULES
#endif

#endif  // _USER_CONFIG_OVERRIDE_H_

compiler:
gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0

So nothing "exotic" in my opinion. But remember, I just asked about the exception message.
Meanwhile a backup system is running and I will update the original one when there is some more time left.

**Reply by Jason2866** [reply]
The exception message is just an indicator. If it is reproducible the same it is worth to have a closer look.
If it happens sometimes and with different values it is a hardware problem. Power, flash or whatever.
If a replacement device works fine, it is sure the hardware.

**Comment by dl7ata** [comment]
Thanks Jason2866. The ESP is changed, the new one is running with the latest firmware so everything is fine.
